azure-linux.dockerfile

```diff
@@ -10,5 +10,8 @@ FROM $IMAGE
 # jq: It's widely used for parsing JSON output in Azure CLI and has a small size. See https://github.com/Azure/azure-cli/issues/29830
 RUN --mount=type=bind,target=/azure-cli.rpm,source=./docker-temp/azure-cli.rpm tdnf install ca-certificates jq /azure-cli.rpm -y && tdnf clean all && rm -rf /var/cache/tdnf
 
+# See https://github.com/Azure/azure-cli/issues/29828 for background on this
+ENV AZ_BICEP_GLOBALIZATION_INVARIANT=1
```

Comment:
`ENV AZ_INSTALLER=DOCKER` is sufficient to detect the Docker environment.

Reply:
While that's technically true, I think it's clearer to define in a separate environment variable. If someone wants to build an image on top of this one with `libicu` installed, they would just need to unset `AZ_BICEP_GLOBALIZATION_INVARIANT`.

There may also be scenarios outside of this container image where it's useful for someone to be able to set this env variable.

Or we may want to distinguish between a "mariner-based docker image" and a "non mariner-based docker image", and change the value of `AZ_BICEP_GLOBALIZATION_INVARIANT` but set `AZ_INSTALLER` to `DOCKER` in both cases.